yt_dlp/extractor/kukululive.py

```diff
@@ -0,0 +1,140 @@
+    int_or_none,
+    join_nonempty,
+    js_to_json,
+    qualities,
+    url_or_none,
+    urljoin,
+)
+from ..utils.traversal import traverse_obj
+
+
+class KukuluLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://live\.erinn\.biz/live\.php\?h(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://live.erinn.biz/live.php?h675134569',
+        'md5': 'e380fa6a47fc703d91cea913ab44ec2e',
+        'info_dict': {
+            'id': '675134569',
+            'ext': 'mp4',
+            'title': 'プロセカ',
+            'description': 'テストも兼ねたプロセカ配信。',
+            'timestamp': 1702689148,
+            'upload_date': '20231216',
+            'thumbnail': r're:^https?://.*',
+        },
+    }, {
+        'url': 'https://live.erinn.biz/live.php?h102338092',
+        'md5': 'dcf5167a934b1c60333461e13a81a6e2',
+        'info_dict': {
+            'id': '102338092',
+            'ext': 'mp4',
+            'title': 'Among Usで遊びます！！',
+            'description': 'VTuberになりましたねんねこ㌨ですよろしくお願いします',
+            'timestamp': 1704603118,
+            'upload_date': '20240107',
+            'thumbnail': r're:^https?://.*',
+        },
+    }, {
+        'url': 'https://live.erinn.biz/live.php?h878049531',
+        'only_matching': True,
+    }]
+
+    def _get_quality_meta(self, video_id, desc, code, force_h264=''):
+        description = desc if force_h264 == '' else f'{desc} (force_h264)'
+        qs = self._download_webpage(
+            'https://live.erinn.biz/live.player.fplayer.php', video_id,
+            query={
+                'hash': video_id,
+                'action': f'get{code}liveByAjax',
+                'force_h264': force_h264,
+            },
+            note=f'Downloading {description} quality metadata',
+            errnote=f'Unable to download {description} quality metadata')
+        return urllib.parse.parse_qs(qs)
+
+    def _add_quality_formats(self, formats, quality_meta):
+        vcodec = traverse_obj(quality_meta, ('vcodec', 0, {str}))
+        quality = traverse_obj(quality_meta, ('now_quality', 0, {str}))
+        quality_priority = qualities(('low', 'h264', 'high'))(quality)
+        if traverse_obj(quality_meta, ('hlsaddr', 0, {url_or_none})):
+            formats.append({
+                'format_id': quality,
+                'url': quality_meta['hlsaddr'][0],
+                'ext': 'mp4',
+                'vcodec': vcodec,
+                'quality': quality_priority,
+            })
+        if traverse_obj(quality_meta, ('hlsaddr_audioonly', 0, {url_or_none})):
+            formats.append({
+                'format_id': join_nonempty(quality, 'audioonly'),
+                'url': quality_meta['hlsaddr_audioonly'][0],
+                'ext': 'm4a',
+                'vcodec': 'none',
+                'quality': quality_priority,
+            })
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        html = self._download_webpage(url, video_id)
+
+        if 'タイムシフトが見つかりませんでした。' in html:
```

Comment:
make sure this isn't added by a user
```suggestion
        if '>タイムシフトが見つかりませんでした。<' in html:
```